Apex7000_BillValidator/Serial/BaseDataValidator.cs

```diff
@@ -0,0 +1,73 @@
+﻿ namespace PyramidNETRS232.Serial
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    
+    /// <summary>
+    /// Base class for defining serial data validation
+    /// </summary>
+    public abstract class BaseDataValidator
+    {
+        /// <summary>
+        /// Creates a new Base Data Validator
+        /// </summary>
+        /// <param name="rawData">Raw data received</param>
+        protected BaseDataValidator(byte[] rawData)
```

Comment:
RawData and Data should be immutable so just a setter is fine. Then called the validation check and set the IsValid flag. That way, every call to the IsValid property doesn't cause a complete recalculation.